#Case Study on Measures of Central Tendency and Dispersion  of Students performance dataset

In [ ]:
import pandas as pd

In [20]:
studentdata='StudentsPerformance.csv'
df=pd.read_csv(studentdata)
df

gender race/ethnicity parental level of education         lunch  \
0    female        group C            some high school  free/reduced   
1    female        group B                 high school  free/reduced   
2    female        group B            some high school  free/reduced   
3    female        group B                some college      standard   
4    female        group C                some college  free/reduced   
..      ...            ...                         ...           ...   
995    male        group E                some college      standard   
996    male        group A                some college      standard   
997    male        group D                some college      standard   
998    male        group E          associate's degree  free/reduced   
999    male        group E           bachelor's degree      standard   

    test preparation course  math score  reading score  writing score  
0                      none           0             17             10  
1                      none           8             24             23  
2                      none          18             32             28  
3                      none          11             38             32  
4                      none          22             39             33  
..                      ...         ...            ...            ...  
995               completed          99             87             81  
996               completed         100             96             86  
997               completed         100             97             99  
998               completed         100            100             93  
999               completed         100            100            100  

[1000 rows x 8 columns]

In [12]:
df.isnull().sum()


gender                         0
race/ethnicity                 0
parental level of education    0
lunch                          0
test preparation course        0
math score                     0
reading score                  0
writing score                  0
dtype: int64

#1 NUMBER OF MALES AND FEMALES PARTICIPATED IN THE TEST

In [13]:
studentdata='StudentsPerformance.csv'
df=pd.read_csv(studentdata)
Gender = df["gender"].value_counts()
Gender

female    518
male      482
Name: gender, dtype: int64

inference:518 females participated in test
          482 males participated in test 

#2 STUDY OF STUDENT'S PARENTAL LEVEL OF EDUCATION

In [8]:
df['parental level of education'].value_counts().to_frame().reset 


index  parental level of education
0        some college                          226
1  associate's degree                          222
2         high school                          196
3    some high school                          179
4   bachelor's degree                          118
5     master's degree                           59

INFERENCE:MOST OF THE PARENTS  EDUCATED IN SOME COLLEGE

#3(a) STUDENTS SCORED ABOVE AVERAGE FOR MATH ,READING,WRITING BASED ON GENDER

In [3]:
import pandas as pd
studentdata='StudentsPerformance.csv'
df=pd.read_csv(studentdata)


In [5]:
df1 = df[['gender', 'math score', 'reading score', 'writing score']].groupby('gender').mean()

In [6]:
df1

math score  reading score  writing score
gender                                          
female   63.376448      72.590734      72.467181
male     68.821577      65.545643      63.446058

INFERENCE:females have higher avertage in reading score and writing score,while male students have high average in math score.

#3(a) STUDENTS SCORED ABOVE AVERAGE FOR MATH ,READING,WRITING BASED ON TEST PREPARATION COURSE

In [9]:
df2= df[['test preparation course', 'math score', 'reading score', 'writing score']].groupby('test preparation course').mean()

In [10]:
df2

math score  reading score  writing score
test preparation course                                          
completed                  69.96648      74.175978      74.684358
none                       63.78972      66.417445      64.457944

INFERENCE:HIGH AVERAGE ON MATH SCORE,READING SCORE,WRITING SCORE FOR THOSE WHO COMPLETED TEST PREPARATION COURSE.

#4(a) THE SCORING VARIATION FOR MATH,READING AND WRITING BASED ON GENDER

In [13]:
genvar= df[['gender', 'math score', 'reading score', 'writing score']].groupby('gender').std()

In [14]:
genvar

math score  reading score  writing score
gender                                          
female   16.029928      14.411018      14.844842
male     14.556411      14.149594      14.227225

# males have less variations in each of the scores.

#4(b) THE SCORING VARIATION FOR MATH,READING AND WRITING BASED ON TEST PREPARATION COURSE

In [15]:
testvar= df[['test preparation course', 'math score', 'reading score', 'writing score']].groupby('test preparation course').std()

In [16]:
testvar

math score  reading score  writing score
test preparation course                                          
completed                 14.521847      13.537572      13.236412
none                      15.705689      14.608896      15.041667

In [ ]:
Inference:Those who completed test preparation course has less variations in score

#5)TOP 25% OF THE STUDENTS BASED ON MATH SCORE

In [30]:
df.nlargest(250, ['math score'])

gender race/ethnicity parental level of education         lunch  \
515  female        group E                some college      standard   
516  female        group E           bachelor's degree      standard   
517  female        group E          associate's degree      standard   
996    male        group A                some college      standard   
997    male        group D                some college      standard   
..      ...            ...                         ...           ...   
428  female        group D                some college  free/reduced   
429  female        group E            some high school      standard   
849    male        group A           bachelor's degree      standard   
850    male        group C          associate's degree  free/reduced   
851    male        group D                some college      standard   

    test preparation course  math score  reading score  writing score  
515                    none         100             92             97  
516                    none         100            100            100  
517                    none         100            100            100  
996               completed         100             96             86  
997               completed         100             97             99  
..                      ...         ...            ...            ...  
428                    none          77             86             86  
429                    none          77             79             80  
849                    none          77             67             68  
850                    none          77             67             64  
851               completed          77             62             62  

[250 rows x 8 columns]

In [ ]:
INFERENCE:LIST OF TOP 250 STUDENTS WHO SCORED HIGH MATHSCORE AND ARE ELIGIBLE FOR BONUS POINTS.